File: sree-core/src/main/java/io/github/sree/information/listeners/PlayerDeathListener.java
package io.github.sree.information.listeners;

import io.github.sree.information.InformationHandler;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;

public class PlayerDeathListener implements Listener {

    private final InformationHandler informationHandler;

    public PlayerDeathListener(InformationHandler informationHandler) {
        this.informationHandler = informationHandler;
    }

    @EventHandler
    public void onPlayerDeath(PlayerDeathEvent event) {
        informationHandler.handleDeathMessage(event);
    }
}

File: soulswap/src/main/kotlin/io/github/sree/soulswap/listeners/PlayerDeathListener.kt
package io.github.sree.soulswap.listeners

import io.github.sree.soulswap.GameManager
import io.github.sree.soulswap.state.GameState
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

internal class PlayerDeathListener(
    val gameManager: GameManager,
    val gameState: GameState
) : Listener {

    @EventHandler
    fun onPlayerDeath(event: PlayerDeathEvent) {
        if (gameState.gameStarted) gameManager.handlePlayerDeath(event)
    }
}